### PR TITLE
Add Project.supported_platforms method

### PR DESCRIPTION
**Pull request by 0xced**
- [x] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

---
Single-scheme projects may support multiple platforms, as explained in [Multiplatform, Single-scheme Xcode Projects](http://promisekit.org/news/2016/08/Multiplatform-Single-Scheme-Xcode-Projects/)

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/project.rb [hunk 1]
```diff
@@ -173,21 +173,25 @@ def app_name
...
 
     def ios?
-      !mac? && !tvos?
+      supported_platforms.include?(:iOS)
+    end
+
+    def supported_platforms
+      supported_platforms = build_settings(key: "SUPPORTED_PLATFORMS").split
+      platforms = []
+      platforms << :macOS   if supported_platforms.include?("macosx")
```
Comment (milch) [inline fastlane_core/lib/fastlane_core/project.rb]: What about the others that were previously in the `mac?` method? Or does the `SUPPORTED_PLATFORMS` key always just contain `macosx`? 

```
return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "macOS"
return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "OS X"
```

Comment (0xced) [inline fastlane_core/lib/fastlane_core/project.rb]: Yes. The _display_ name changed with Xcode versions but the underlying platform never changed, just like iOS platform is still `iphoneos` and `iphonesimulator` even though iPhone OS was renamed iOS 6 years ago.

File: fastlane_core/lib/fastlane_core/project.rb [hunk 2]
```diff
@@ -173,21 +173,25 @@ def app_name
...
+      platforms << :iOS     if (supported_platforms & ["iphonesimulator", "iphoneos"]).any?
+      platforms << :watchOS if (supported_platforms & ["watchsimulator", "watchos"]).any?
+      platforms << :tvOS    if (supported_platforms & ["appletvsimulator", "appletvos"]).any?
+      return platforms
     end
```
Comment (mfurtak) [inline fastlane_core/lib/fastlane_core/project.rb]: Working around Ruby's lack of Sets is always a pain... What do you think about this:

```ruby
supported_platforms = build_settings(key: "SUPPORTED_PLATFORMS").split
supported_platforms.reduce([]) do |acc, val|
  acc << case val
         when "macosx" then :macOS
         when "iphonesimulator", "iphoneos" then :iOS
         when "watchsimulator", "watchos" then :watchOS
         when "appletvsimulator", "appletvos" then :tvOS
         end
end.uniq!
```

**Comment by mfurtak** [comment]
Hi @0xced - there was a change to `faraday` that has caused some of our tests to break. This is fixed on master, now. Could you please rebase onto that to get CI happy again? Thanks!

**Review by mfurtak — APPROVED** [review]
Thanks for the improvement! 👍

**Pull request merged**